replication/client_lib/src/rrdb_client_factory_impl.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include "rrdb_client.h"
+#include "rrdb_error.h"
+#include "rrdb_client_impl.h"
+
+extern void dsn_core_init();
+
+namespace dsn{ namespace apps{
+
+class rrdb_client_factory_impl{
+public:
+    static bool initialize(const char* config_file);
+
+    static irrdb_client* get_client(const char* app_name, const char* cluster_name);
+
+private:
+    typedef std::unordered_map<std::string, rrdb_client_impl*> app_to_client_map;
+    static std::unordered_map<std::string, app_to_client_map> _cluster_to_clients;
```

Comment:
(1) use cluster_name + app_name as the string key can simplify the maps
(2) it is best to use single-ton instead of lots of static for the methods. 
(3) WARNING: as we use static or single-ton, this class instance will be used by multiple apps if there are multiple apps in the process, which makes simulation impossible as the lock etc. now belongs to multiple apps.